standard_sequence/classy_sequence.py

```diff
@@ -861,263 +318,528 @@ def _do_molasses_tof_sequence(self, t, reset_mot = False):
-                                       shot_globals.odp_repump_power,)
-            # Need to turn off the TA before repump, Sam claims this timing should work
-            assert shot_globals.odp_ta_time > shot_globals.odp_repump_time, "TA time should be longer than repump for depumping to F = 3"
+            t += max(D2Lasers.CONST_TA_VCO_RAMP_TIME, shot_globals.op_ramp_delay)
+            t, t_aom_start = self.D2Lasers_obj.do_pulse(
+                t,
+                shot_globals.odp_ta_time,
+                ShutterConfig.OPTICAL_PUMPING_FULL,
+                shot_globals.odp_ta_power,
+                shot_globals.odp_repump_power,
+                close_all_shutters = close_all_shutters
+            )
+
+            t_aom_off = t_aom_start + shot_globals.odp_ta_time
+
+            assert (
+                shot_globals.odp_ta_time > shot_globals.odp_repump_time
+            ), "TA time should be longer than repump for depumping to F = 3"
             # TODO: test this timing
-            self.D2Lasers_obj.ta_aom_off(t + shot_globals.odp_ta_time - shot_globals.odp_repump_time)
-            # Close the shutters
-            self.D2Lasers_obj.update_shutters(t, ShutterConfig.NONE)
-            t += CONST_SHUTTER_TURN_OFF_TIME
+            self.D2Lasers_obj.repump_aom_off(
+                t_aom_start + shot_globals.odp_repump_time
+            )
 
-            return t
 
-        else:
-            raise NotImplementedError("This optical depumping method is not implemented")
+            return t, t_aom_off
 
+        else:
+            raise NotImplementedError(
+                "This optical depumping method is not implemented"
+            )
 
-    def kill_F4(self, t):
-        ''' Push away atoms in F = 4 '''
+    def kill_F4(self, t, shutter_config = ShutterConfig.OPTICAL_PUMPING_TA, close_all_shutters = True):
+        """Push away atoms in F = 4"""
         # tune to resonance
-        self.D2Lasers_obj.ramp_ta_freq(t, 0, 0)
-        t += CONST_TA_VCO_RAMP_TIME
+        self.D2Lasers_obj.ramp_ta_freq(t, D2Lasers.CONST_TA_VCO_RAMP_TIME, shot_globals.killing_pulse_detuning)
+        t += D2Lasers.CONST_TA_VCO_RAMP_TIME
         # do a ta pulse via optical pumping path
-        t = self.D2Lasers_obj.do_pulse(t, shot_globals.op_killing_killing_pulse_time,
-                                    ShutterConfig.OPTICAL_PUMPING_FULL,
-                                    shot_globals.op_killing_ta_power, 0, close_all_shutters=True)
+        t, _ = self.D2Lasers_obj.do_pulse(
+            t,
+            shot_globals.op_killing_pulse_time,
+            shutter_config,
+            shot_globals.op_killing_ta_power,
+            0,
+            close_all_shutters=close_all_shutters,
+        )
+
+        return t
 
     def kill_F3(self, t):
         pass
 
+    def _optical_pump_molasses_sequence(self, t, reset_mot=False):
+        # MOT loading time 500 ms
+        mot_load_dur = 0.5
 
+        t += D2Lasers.CONST_SHUTTER_TURN_ON_TIME
```

Comment:
delete this statement? no longer necessary?

Reply:
Deleted